Sources/Sentry/SentrySpanContext.m

```diff
@@ -91,6 +91,9 @@ - (void)removeTagForKey:(NSString *)key
     if (self.operation != nil)
         [mutabledictionary setValue:self.operation forKey:@"op"];
 
+    if (self.description != nil)
+        [mutabledictionary setValue:self.spanDescription forKey:@"description"];
+
     if (self.spanDescription != nil)
         [mutabledictionary setValue:self.spanDescription forKey:@"description"];
 
```

Comment:
h: There is something wrong here. You are checking for self.description (which is always not nil unless overridden), and then copy self.spanDescription to dictionary. 

I guess you want to copy "description", but this property has no setter and returns class name and pointer, thats the reason we have self.spanDescription, it is different from other SDKs, but is a language limitation.


```suggestion
      if (self.spanDescription != nil)
        [mutabledictionary setValue:self.spanDescription forKey:@"description"];

```

Reply:
Good catch, @brustolin 👏 . I wonder why the tests are not failing then.

Reply:
I added more tests for this part of the code in https://github.com/getsentry/sentry-cocoa/pull/1147. It's not failing because the dictionary just doesn't add anything when you add nil to it. Anyway, I removed the code.